.github/actions/setup_nox/action.yml

```diff
@@ -0,0 +1,33 @@
+name: Setup Nox
+description: 'Prepare and cache nox, install poetry.'
+
+inputs:
+  cache_number:
+    description: "A manually specified cache number. Useful for triggering new caches."
+    required: True
+    default: 0
+
+runs:
```

Comment:
Interesting — maybe `test-cargo.yml` should be an action like this, rather than a workflow

Reply:
Could be! I'm a big fan of composite actions - they make things so much more dry.

Reply:
I just changed the cargo one to this design!